### ISSUE TITLE
`Style/HashEachMethods` mistakes other objects which define a `values` method as a Hash

### ISSUE THREAD

**Issue by postmodern** [rubocop/rubocop#12942]
The `Style/HashEachMethods`  cop mistakes other objects which define a `values` reader method, which `.each` is called on it's return value, as a `Hash`.

```ruby
class Container
  attr_reader :values
  attr_reader :others

  def initialize(values, others: [])
    @values = values
    @others = others
  end
end

container = Container.new([1, 2, 3, 4])
container.values.each do |value|
  # ...
end
```

## Expected behavior

Recognize that `container` is not a `Hash`.

## Actual behavior

```
For /home/postmodern/test/ruby/rubocop: Default configuration from /home/postmodern/test/ruby/rubocop/vendor/bundle/ruby/3.2.0/gems/rubocop-1.64.0/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /home/postmodern/test/ruby/rubocop/test.rb
Loading cache from /home/postmodern/.cache/rubocop_cache/1ebcb8f50a98b3b618576efe59148c89e9aa2364/6d7a3b621ca1730e04accd938619e4bdab66cfb1/a2ce522b6457b198fbca54a1a1551536b82162dd

test.rb:10:11: C: [Correctable] Style/HashEachMethods: Use each_value instead of values.each.
container.values.each do |value|
          ^^^^^^^^^^^
```

## Steps to reproduce the problem

`Gemfile`:
```ruby
source 'https://rubygems.org'

gem 'rubocop'
```

`test.rb`
```ruby
class Container
  attr_reader :values

  def initialize(values)
    @values = values
  end
end

container = Container.new([1, 2, 3, 4])
container.values.each do |value|
  # ...
end
```

```shell
$ bundle exec rubocop test.rb
```

## RuboCop version

```
1.64.0 (using Parser 3.3.1.0, rubocop-ast 1.31.3, running on ruby 3.2.4) [x86_64-linux]
```

**Comment by koic** [comment]
This cop is already marked as unsafe:
https://docs.rubocop.org/rubocop/1.64/cops_style.html#stylehasheachmethods

It can be avoided by using the `# rubocop:disable` directive comment, `AllowedReceiver` parameter, or others.

**Comment by postmodern** [comment]
@koic shouldn't this cop be disabled by default if it is unsafe?
